builder/virtualbox/common/vboxmanage_config.go

```diff
@@ -11,7 +15,25 @@ type VBoxManageConfig struct {
 func (c *VBoxManageConfig) Prepare(ctx *interpolate.Context) []error {
 	if c.VBoxManage == nil {
 		c.VBoxManage = make([][]string, 0)
+		return nil
+	}
+
+	var errs []error
+	var err error
+	var desiredMem uint64
+
+	for _, cmd := range c.VBoxManage {
+		if cmd[2] == "--memory" {
+			desiredMem, err = strconv.ParseUint(cmd[3], 10, 64)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Error parsing string: %s", err))
+			}
+		}
+	}
+
+	if err = common.AvailableMem(desiredMem); err != nil {
```

Comment:
Since this is in `common` I'm curious what happens when packer builds for ESXi where the remote host has more resources than the machine running packer. Does this cause the build to fail when the local machine has insufficient resources even if the ESXi host has enough resources?

We could probably test this by running packer in a VM with 512mb ram against ESXi in a VM with 1gb ram (for example).